Instructions for installing from OperatorHub (#37)
skupperproject/skupper-operator: README.md

# Installing the skupper-operator

The instructions below demonstrate how to install the skupper-operator
and run it inside the `my-namespace` namespace. If you want to install it on
a different namespace, edit the referenced yaml before applying.

If you want to test your catalog against a local minikube cluster,
you'll need to install OLM first. For more info, check this out:
https://olm.operatorframework.io/docs/getting-started/

In an OpenShift cluster, OLM is already installed.

# Installing from OperatorHub.io (Community Operators)

Skupper Operator is also available on [OperatorHub.io](https://operatorhub.io/operator/skupper-operator).

> If the `Community Operators` catalog is available in your cluster, you

Reviewer: ```suggestion
You
```
I don't think other projects talk about 'Community Operators' catalog. What's the purpose of putting in this pre-req?

Author: If you already have the `Community Operators` catalog available in your cluster, then you don't need
to install our own catalog source.

> can install the Skupper Operator by visiting the link above and clicking the `Install` button,

Reviewer: Given that the `Install` button only displays doc and that doc only repeats what's below, but adds  mentions  of CRDs, maybe best to not draw attention to 'Install' button?

following the respective instructions (only for Kubernetes).

> 

Reviewer: ```suggestion

The installation documentation mentions `To use it, checkout the custom resource definitions (CRDs) introduced by this operator to start using it`.
However, there are no CRDs associated with the Skupper Operator, continue reading this documentation to start using it.

```

Reviewer: The PR looks good, but I still think there's an issue with operatorhub because when you click **Install** you see
![image](https://github.com/skupperproject/skupper-operator/assets/5154224/78c5d25e-c27b-4f1f-a60c-0aa581a664fa)

Reviewer: ah, just realised you remove the reference to Install button, that's enough

This procedure installs the Skupper Operator at cluster level (watching all namespaces).

> Usually to install the latest published version of the Skupper Operator on Kubernetes (only), run:

Reviewer: ```suggestion
To install the latest published version of the Skupper Operator on a Kubernetes cluster where OLM is installed, run:
```
Not sure what you mean by 'Kubernetes (only)'?

Author: Non-Openshift.

```shell
kubectl create -f https://operatorhub.io/install/skupper-operator.yaml
```

> Verify it is up and running executing:

Reviewer: ```suggestion
Verify the Skupper Operator is running by executing:
```

```shell
kubectl get csv -n operators
```

> If you want to install it to a single namespace, follow the `Installing on Minikube` or

Reviewer: ```suggestion
If you want to install the Skupper Operator in a single namespace, follow the `Installing on Minikube` or
```

> `Installing on OpenShift` sections.

Reviewer: ```suggestion
`Installing on OpenShift` sections below.
```

Otherwise, after the operator has been installed from OperatorHub.io, you can skip the following sections
and go straight to [Creating a new skupper site](#creating-a-new-skupper-site)

## Installing on Minikube

To install the Skupper Operator CatalogSource on your Minikube cluster, run:

```

> # Creating a CatalogSource on the olm namespace

Reviewer: not sure what prompted you to change from 'in' to 'on', but I think it should be 'in'
eg https://docs.openshift.com/container-platform/4.14/support/troubleshooting/troubleshooting-operator-issues.html

kubectl apply -f examples/k8s/00-cs.yaml

# Wait for the skupper-operator catalog pod to be running
kubectl -n olm get pods | grep skupper-operator
```

Once the catalog is available, you can then create the subscription:

```
# Create an OperatorGroup on the `my-namespace` namespace
kubectl apply -f examples/k8s/10-og.yaml

# Create a Subscription on the `my-namespace` namespace
kubectl apply -f examples/k8s/20-sub.yaml
```

## Installing on OpenShift

To install the Skupper Operator CatalogSource on your OpenShift cluster, run:

```

> # Creating a CatalogSource on the `openshift-marketplace` namespace

Reviewer: as above
